quipucords/tests/api/reports/test_reports.py

```diff
@@ -290,3 +290,75 @@ def test_report_deployments_csv(django_client, deployment_with_logs: Deployments
     # Why 7? We expect two separate table headers in the file, one section marker,
     # three blank lines, and at least one line per table. Yes, this CSV is a mess. >:(
     assert len(tarball_deployments_csv_lines) >= 7
+
+
+def test_report_details_consistency(
+    django_client, deployment_with_logs: DeploymentsReport
+):
+    """Test report details response is driven by request headers and not cached."""
+    deployments_report = deployment_with_logs
+    report = deployments_report.report
+
+    for _ in range(2):
```

Comment:
I don't understand why we are running this code twice. Both iterations use the same report IDs, and both assert the same results. How does this exercise or prove that the response is not cached, assuming that is the intent based on the docstring? Am I overlooking or misunderstanding something?

Reply:
The Camayoc test for that traces to https://github.com/quipucords/camayoc/issues/195 from 2018, with following rationale:

> Test that you receive the appropriate content back when specifying a content type.
> Test that after you selected both content types that caching doesn't cause any side-effects and you get the appropriate content type back.

I wasn't around back then, so I can only assume this was done in response to actual bug, or a risk considered by a team at the time. And honestly, "check that our caching is not too aggressive" does sound reasonable, especially when it's cheap.

I was on the fence with that one myself, so I can remove this if you think it doesn't provide enough value to keep around.

Reply:
For caching, I think the idea was "if I request JSON and CSV, then I get JSON and CSV respectively". I.e. I don't get back cached CSV content if I request JSON after CSV. It's run twice to ensure cache is loaded (if any) and we cover both "request JSON and then CSV" and "request CSV and then JSON".

Reply:
I keep looking at this, and although I think I understand the historic context you provided, I don't think the current test provides much value. I think my vote would be simply to remove these tests, or at least remove the loop while keeping one-time assertions, assuming the basic behavior is not already covered elsewhere.

I don't feel _super_ strongly one way or the other, and I'll trust your judgement about keeping or removing them. You may resolve this conversation (and the other) whenever you're ready to merge.

Reply:
I removed them in 91f870b128fbac2de83290ee124e0605f6dba0ef (with intent it will be squashed before merge). Since nobody has strong opinion and reason to keep them around, I think we can take a risk.

With recent discussion about changes we want to do around generating and storing reports, hopefully we will get better tests for that anyway.

I'll keep that thread open for a little longer, in case anyone wants to voice their opinion.